internal/graph/graph.go

```diff
@@ -78,6 +103,8 @@ type RelationshipIngress struct {
 	// object reference with a target if the type of the relationship ingress is that
 	// of a TupleToUserset
 	TuplesetRelation *openfgapb.RelationReference
+
+	Condition IngressCondition
```

Comment:
The ingress is a conditional one, so I believe this makes a lot of sense from a modeling perspective. If the ingress is conditional, then that means there was at least one or more other relationships that would otherwise have to be evaluated as part of the intersection or exclusion. 

This relates also to the behavior around pruned relationship ingresses. See https://github.com/openfga/openfga/pull/797#discussion_r1218531400

Reply:
I'll rephrase my comment: If we use `RelationshipIngress` for any other algorithm, will `Condition` still be applicable?

Reply:
Yep, definitely. The ingress is conditioned upon some other relationship ingress, and so any objects reverse-expanded under a conditional ingress must be further evaluated to resolve the intersection/exclusion.